hypothesis/hypo_vector.py

```diff
@@ -0,0 +1,130 @@
+        self.sin_theta_v = math.sin(self.theta_v)
+        self.cos_theta_v = math.cos(self.theta_v)
+        self.sin_phi_v = math.sin(self.phi_v)
+        self.cos_phi_v = math.cos(self.phi_v)
+        self.speed_x = self.speed_of_light * self.sin_theta_v * self.cos_phi_v
+        self.speed_y = self.speed_of_light * self.sin_theta_v * self.sin_phi_v
+        self.speed_z = self.speed_of_light * self.cos_theta_v
+        #convert track energy to length
+        self.trck_length = 15. / 3.3 * trck_energy
+        #precalculated (nphotons.py) to avoid icetray
+        self.photons_per_meter = 2451.4544553
+        self.cscd_photons = 12805.3383311 * cscd_energy
+        self.trck_photons = self.trck_length * self.photons_per_meter
+        self.tot_photons = self.cscd_photons + self.trck_photons
+
+    
+    def set_binning(self, n_t_bins, n_r_bins, n_theta_bins, n_phi_bins, t_max, r_max, t_min=0, r_min=0):
+        '''
+        define binnings of spherical coordinates
+        t_min : min time (ns)
+        t_max : max time (ns)
+        r_min : min radius (m)
+        r_max : max radius (m)
+        '''
+        self.n_t_bins = n_t_bins
+        self.n_r_bins = n_r_bins
+        self.n_theta_bins = n_theta_bins
+        self.n_phi_bins = n_phi_bins
+        self.t_min = t_min * 1e-9
+        self.t_max = t_max * 1e-9
+        self.r_min = r_min
+        self.r_max = r_max
+        self.t_scaling_factor = self.n_t_bins / (self.t_max - self.t_min)
+        self.r_scaling_factor = self.n_r_bins * self.n_r_bins / self.r_max
+        self.theta_scaling_factor = self.n_theta_bins / 2.
+        self.phi_scaling_factor = self.n_phi_bins / np.pi / 2.
+        self.phi_bin_width = 2. * np.pi / self.n_phi_bins
+
+    def set_dom_location(self, t_dom=0., x_dom=0., y_dom=0., z_dom=0.):
+        '''
+        changes the track vertex to be relative to a dom at a given position with
+        t_dom : time (ns)
+        x_dom, y_dom, z_dom : position of the dom (m)
+        '''
+        self.t = self.t - t_dom * 1e-9
+        self.x = self.x - x_dom
+        self.y = self.y - y_dom
+        self.z = self.z - z_dom
+
+    def vector_photon_matrix(self):
+        '''
+        uses a single time array to simultaneously calculate all of the positions along the track, using information from __init__
+        '''
+        #create initial time array, using the midpoints of each time increment
+        self.t_array_init = np.arange(self.t, min(self.t_max, self.trck_length / self.speed_of_light + self.t), self.time_increment, dtype=np.float32) - self.time_increment / 2
+        self.t_array_init[0] = self.t
+        #set the number of time increments in the track
+        self.number_of_increments = int(len(self.t_array_init))
+        #create array with variables
+        self.variables_array = np.empty((8, self.number_of_increments), dtype=np.float32)
+        self.t_array = self.variables_array[0, :]
+        self.t_array[:] = self.t_array_init 
```

Comment:
these assignements seem a bit over complicated? why not just refer to them directly as variables_array[0, :]?